src/editor/UI/FileExplorerView.re

```diff
@@ -0,0 +1,43 @@
+open Oni_Model;
+open Oni_Core;
+open Revery_UI;
+open Revery.UI.Components;
+
+let component = React.component("FileExplorerView");
+
+let createElement = (~children, ~state: State.t, ()) =>
+  component(hooks => {
+    open TreeView;
+    let hooks =
+      React.Hooks.effect(
+        OnMount,
+        () => {
+          let cwd = Rench.Environment.getWorkingDirectory();
+          GlobalContext.current().dispatch(OpenExplorer(cwd));
+          None;
+        },
+        hooks,
+      );
+
+    let onNodeClick = (clicked, tree) => {
+      Tree.(
+        switch (clicked) {
+        | Node({data}, _) =>
+          let node = FileExplorer.toFsNode(data);
+          GlobalContext.current().dispatch(SetExplorerTree(tree));
+          GlobalContext.current().dispatch(OpenFileByPath(node.path));
```

Comment:
Do we want to gate this to run only on `node.isDirectory` is false? Else we would be opening folders like files.